tests/memory_checker/test_memory_checker.py

```diff
@@ -0,0 +1,201 @@
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.assertions import pytest_require
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+CONTAINER_STOP_THRESHOLD_SECS = 200
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+CONTAINER_CHECK_INTERVAL_SECS = 1
+
+
+@pytest.fixture(autouse=True, scope="module")
+def modify_monit_config_and_restart(duthost):
+    """Backup Monit configuration file, then customize and restart it before testing. Restore original
+    Monit configuration file and restart it after testing.
+
+    Args:
+        duthost: Hostname of DuT.
+
+    Returns:
+        None.
+    """
+    logger.info("Back up Monit configuration file ...")
+    duthost.shell("sudo cp -f /etc/monit/monitrc /tmp/")
+
+    logger.info("Modifying Monit config to eliminate start delay and decrease interval ...")
+    duthost.shell("sudo sed -i 's/set daemon 60/set daemon 10/' /etc/monit/monitrc")
+    duthost.shell("sudo sed -i '/with start delay 300/s/^./#/' /etc/monit/monitrc")
+
+    logger.info("Restart Monit service ...")
+    duthost.shell("sudo systemctl restart monit")
+
+    yield
+
+    logger.info("Restore original Monit configuration ...")
+    duthost.shell("sudo mv -f /tmp/monitrc /etc/monit/")
+
+    logger.info("Restart Monit service ...")
+    duthost.shell("sudo systemctl restart monit")
+
+
+def install_stress_utility(duthost, container_name):
+    """Installs the 'stress' utility in container before testing.
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container_name: Name of container.
+
+    Retuerns:
+        None.
+    """
+    logger.info("Installing 'stress' utility in '{}' container ...".format(container_name))
+
+    install_cmd_result = duthost.shell("docker exec {} bash -c 'export http_proxy=http://100.127.20.21:8080 \
+                                        && export https_proxy=http://100.127.20.21:8080 \
+                                        && apt-get install stress -y'".format(container_name))
+
+    exit_code = install_cmd_result["rc"]
+    pytest_assert(exit_code == 0, "Failed to install 'stress' utility!")
+    logger.info("'stress' utility was installed.")
+
+
+def remove_stress_utility(duthost, container_name):
+    """Removes the 'stress' utility from container after testing.
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container_name: Name of container.
+
+    Retuerns:
+        None.
+    """
+    logger.info("Removing 'stress' utility on device ...")
+    remove_cmd_result = duthost.shell("docker exec {} apt-get remove stress -y".format(container_name))
+    exit_code = remove_cmd_result["rc"]
+    pytest_assert(exit_code == 0, "Failed to remove 'stress' utility!")
+    logger.info("'stress' utility was removed.")
+
+
+def eatup_memory(duthost, container_name):
+    """Eats up more than 400MB memory in specified container.
```

Comment:
Can/should we make the amount of memory to consume configurable, like via passing a parameter? We could then extend this test to other containers besides telemetry.

Reply:
Very good suggestion! Currently at here the parameter `-m` of `stress` utility is hard coded for telemetry container. What I am thinking is that we can configure this parameter according to the memory threshold value of each container. But let me update to pass this parameter as an argument of this function.